standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/messaging/MessageFactory.java

```diff
@@ -63,39 +65,36 @@
 
   private static MessageFactory instance = null;
 
-  protected static final HiveConf hiveConf = new HiveConf();
+  protected static final Configuration conf = MetastoreConf.newMetastoreConf();
+  /*
+  // TODO MS-SPLIT I'm 99% certain we don't need this, as MetastoreConf.newMetastoreConf already
+  adds this resource.
   static {
-    hiveConf.addResource("hive-site.xml");
+    conf.addResource("hive-site.xml");
   }
+  */
 
-  // This parameter is retained for legacy reasons, in case someone implemented custom
-  // factories. This, however, should not be the case, since this api was intended to
-  // be internal-only, and we should manage the jms and json implementations without
-  // needing this parameter. Marking as deprecated, for removal by 2.4 - see corresponding
-  // note on the getDeserializer(String,String) method
-  @Deprecated
-  private static final String CONF_LABEL_HCAT_MESSAGE_FACTORY_IMPL_PREFIX = "hcatalog.message.factory.impl.";
-
-  protected static final String MS_SERVER_URL = hiveConf.get(HiveConf.ConfVars.METASTOREURIS.name(), "");
-  protected static final String MS_SERVICE_PRINCIPAL = hiveConf.get(HiveConf.ConfVars.METASTORE_KERBEROS_PRINCIPAL.name(), "");
+  protected static final String MS_SERVER_URL = MetastoreConf.getVar(conf, ConfVars.THRIFT_URIS, "");
+  protected static final String MS_SERVICE_PRINCIPAL =
+      MetastoreConf.getVar(conf, ConfVars.KERBEROS_PRINCIPAL, "");
 
   /**
    * Getter for MessageFactory instance.
    */
   public static MessageFactory getInstance() {
     if (instance == null) {
       instance =
-          getInstance(hiveConf.get(HiveConf.ConfVars.METASTORE_EVENT_MESSAGE_FACTORY.varname));
+          getInstance(MetastoreConf.getVar(conf, ConfVars.EVENT_MESSAGE_FACTORY));
     }
     return instance;
   }
 
   private static MessageFactory getInstance(String className) {
     try {
-      return (MessageFactory)ReflectionUtils.newInstance(JavaUtils.loadClass(className), hiveConf);
+      return JavaUtils.newInstance(JavaUtils.getClass(className, MessageFactory.class));
     }
-    catch (ClassNotFoundException classNotFound) {
-      throw new IllegalStateException("Could not construct MessageFactory implementation: ", classNotFound);
+    catch (MetaException e) {
```

Comment:
JavaUtils.newInstance throws RunTimeException which will not be caught by catching MetaException. Is this intentially catching metaException? If so, when will that be thrown?

Reply:
JavaUtils.getClass throws a MetaException.